File: leetcodely/python/check_completeness_of_binary_tree.py
from collections import deque


class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def isCompleteTree(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        seen_null = False
        queue = deque()
        queue.appendleft(root.left)
        queue.appendleft(root.right)

        while len(queue) > 0:
            curr = queue.pop()
            if curr is None:
                seen_null = True
            else:
                if seen_null:
                    return False
                queue.appendleft(curr.left)
                queue.appendleft(curr.right)

        return True

File: container2/leetcodely/javascript/src/checkCompletenessOfBinaryTree.js
const isCompleteTree = root => {
    const queue = [root];
    let isLast = false;

    while (queue.length > 0) {
        const node = queue.shift();

        if (node.left) {
            if (isLast) return false;
            queue.push(node.left);
        } else {
            isLast = true;
        }

        if (node.right) {
            if (isLast) return false;
            queue.push(node.right);
        } else {
            isLast = true;
        }
    }

    return true;
};